src/vni-manager.js

```diff
@@ -0,0 +1,158 @@
+ *
+ *     node.vni().errorState( state );
+ *     var state = node.vni().errorState();
+ *
+ *     node.vni().inputStates( {input: state} );
+ *     var inputStates = node.vni().inputStates();
+ *
+ *     node.vni().outputState( state );
+ *     var state = node.vni().outputState();
+ */
+module.exports = function( componentFacade ) { 
+
+    _.extend( componentFacade, 
+        {
+
+          /**
+           * Delete all vnis assiciated with the node instance.
+           *
+           * @this node instance
+           * @return node instance node for easy chaining
+           */ 
+          deleteAllVnis: function() { 
+              // reassign vnis array to an empty array and let garbage collection clean up
+              this.vnis = {};
+              return this;
+          },
+
+          /**
+           * Delete the vni associated with the specified VNID
+           * 
+           * @param vnid vnid identifying which vni to delete
+           *
+           * @this node instance
+           * @return node instance node for easy chaining
+           */ 
+          deleteVni: function( vnid ) {
+
+              if ( vnid in this.vnis ) {
+                  delete this.vnis[vnid];
+              } 
+
+              return this;
+          },
+
+          /**
+           * Getter for a vni facade specified by the vnid parameter.  If no VNID is specified,
+           * the IIP VNID will be used.  If the VNI does not yet exist, a new VNI will be
+           * created.
+           * 
+           * @this node 
+           * @param vnid vnid identifying which vni to delete
+           *
+           * @return node for easy chaining
+           */ 
+          vni: function( vnid ) { 
+
+              vnid = vnid || DEFAULT_VNID;
+
+              if ( _.isUndefined( this.vnis[vnid] )  ) {
+
+                  // have no vni so create one
+                  this.vnis[vnid] = {}; 
+
+                  // TODO: Add parentVni setting here
+                  // TODO: Add previousLms here
+              }
+
+              return {
+                  delete: _.bind( this.deleteVni, this, vnid ),
+                  inputStates: _.partial( inputStates, this, vnid ), 
+                  errorState: _.partial( errorState, this, this.vnis[vnid] ), 
+                  outputState: _.partial( outputState, this, this.vnis[vnid] ) 
+              };
+          },
+
+          vnis: {}
+
+    }); // extend
+} // module export
+
+/**
+ * Get/Set error state on the vni 
+ * 
+ * @this vni context 
+ *
+ * @param node current node instance
+ * @param vni vni whose state should be set
+ * @param state error state to be set
+ */
+function errorState( node, vni, state ) {
+
+    // Do we have a state to be set ?
+    if ( arguments.length > 2 ) {
+
+        if ( _.isUndefined(state) ) { 
+            // Clear state
+            delete vni.errorState;
+        } else { 
+            // Set state
+            vni.errorState = state;
+        }
+
+        return node;
```

Comment:
I think this should be 'return this;' so that it returns the vni facade and not the node facade. Same goes for outputState below.